modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/introspection/RuntimeSnapshot.java

```diff
@@ -338,19 +338,19 @@ public ServiceInfo getService(String id) {
 
     @Override
     public List<String> getJavaComponentIds() {
-        return javaComponentsIds;
+        return javaComponentsIds.stream()
+                                .sorted()
+                                .collect(Collectors.collectingAndThen(Collectors.toList(),
+                                        Collections::unmodifiableList));
```

Comment:
By importing statically `collectingAndThen` and `toList` you can have a one-line for this method.

Reply:
Even better: since Java 10 there is `Collectors.toUnmodifiableList()`